sql: fix the asyncpg tests to hit cockroach

ZhouXing19 opened the pull request:
The current asyncpg nightly tests are not run against the cockroach process,
but against an automatically started PostgreSQL process.

This commit removes the step to download PostgreSQL, and runs the tests against
cockroachdb. It also adds failed tests into the blocklist for asyncpg.

Release note (sql): fix the asyncpg tests to hit cockroach

otan reviewed (commented):
how many tests do we pass :o

ZhouXing19 commented on pkg/cmd/roachtest/tests/asyncpg_blocklist.go:
```diff
@@ -0,0 +1,120 @@
+	{"v20.2", "asyncpgBlocklist20_2", asyncpgBlocklist20_2, "asyncpgIgnoreList20_2", asyncpgIgnoreList20_2},
+	{"v21.1", "asyncpgBlocklist21_1", asyncpgBlocklist21_1, "asyncpgIgnoreList21_1", asyncpgIgnoreList21_1},
+	{"v21.2", "asyncpgBlocklist21_2", asyncpgBlocklist21_2, "asyncpgIgnoreList21_2", asyncpgIgnoreList21_2},
+	{"v22.1", "asyncpgBlocklist22_1", asyncpgBlocklist22_1, "asyncpgIgnoreList22_1", asyncpgIgnoreList22_1},
```
But the current roachtest is running with cockroach v21.1, maybe we should put them for 21_2 onwards?

ZhouXing19 commented:
@otan it passed just few :(
```
Tests run against asyncpg v0.24.0
135 Total Tests Run
135 tests passed
0 tests failed
34 tests skipped
92 tests ignored
0 tests passed unexpectedly
0 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run

```

otan commented on pkg/cmd/roachtest/tests/asyncpg.go:
```diff
@@ -92,17 +87,34 @@ func registerAsyncpg(r registry.Registry) {
+			version, blocklistName, ignoredlistName)
+
+		t.Status("Running asyncpg tests ")
+		rawResults, _ := c.RunWithBuffer(
```
should we log the error from this anyway?

ZhouXing19 replied:
Not sure if I understand it correctly -- the error is already logged in one of the `cd.log` under `artifacts/asyncpg/run_1/`. [Here's an example](https://gist.github.com/ZhouXing19/82211c66aa2555df3af4f84103f79b6b).

otan replied:
running the command can fail for other reasons, so i'd prefer if something happens we still have a record.

```
rawResults, err := c.RunWithBuffer(...)
if err != nil {
  t.Logf("error during asyncpg run (may be ok): %v\n", err)
}
```

otan reviewed (commented):
kinda wonder if it's worth having this roachtest if we fail ~ half, cc @rafiss

rafiss commented:
> if it's worth having this roachtest if we fail ~ half

yeah it's still worth it for a few reasons:
- we can establish a baseline number of errors, and work towards improving the passing count in a way that's tracked in git history
- we can start filling in the failure causes to identify and prioritize issues
- we can make sure we don't regress on any other support

otan reviewed (approved):
(lgtm otherwise!)

rafiss commented on pkg/cmd/roachtest/tests/asyncpg.go:
```diff
@@ -45,13 +51,16 @@ func registerAsyncpg(r registry.Registry) {
 
 		t.Status("cloning asyncpg and installing prerequisites")
 
+		t.L().Printf("Latest asyncpg release is %s.", asyncpgSupportedTag)
```
the "latest" line is not accurate. it should try to fetch the latest tag from the repo. for example,  see the hibernate test:
```
		latestTag, err := repeatGetLatestTag(
			ctx, t, "hibernate", "hibernate-orm", hibernateReleaseTagRegex,
		)
```

rafiss commented:
needs a spelling fix to make linting pass
```
/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/asyncpg_blocklist.go:37:80: "cancelled" is a misspelling of "canceled"
```

ZhouXing19 commented:
> needs a spelling fix to make linting pass
> 
> ```
> /go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/asyncpg_blocklist.go:37:80: "cancelled" is a misspelling of "canceled"
> ```

@rafiss seems that it's because the test is named with `cancelled` by asyncpg :D. Changed it to `canceled` will miss this test and give an unexpected fail.

rafiss commented:
lol.. that's funny. we can ignore this file here: https://github.com/cockroachdb/cockroach/blob/82c3d0b4e93d4a6dac6f9c94feaaa2f2199996c3/pkg/testutils/lint/lint_test.go#L1307

rafiss commented:
it looks like you accidentally modified `pkg/sql/lex/BUILD.bazel` -- should be reverted

ZhouXing19 committed:
sql: fix the asyncpg tests to hit cockroach

The current ayncpg nightly tests are not run against the cockroach process,
but against an automatically started postgresql process.

This commit removes the step to download postgresql, and run the tests against
cockroachdb. It also adds failed tests into the blocklist for asyncpg.

Release note (sql): fix the asyncpg tests to hit cockroach

ZhouXing19 commented:
Looks like the CI is failing with a flakey error. @rafiss do you think it’s ok for a merge?

rafiss reviewed (approved):
lgtm!

ZhouXing19 commented:
Thanks!
bors r+
